src/app/gps/qgsgpsinformationwidget.cpp

```diff
@@ -1595,27 +1595,40 @@ void QgsGpsInformationWidget::cursorCoordinateChanged( const QgsPointXY &point )
   try
   {
     mLastCursorPosWgs84 = mCanvasToWgs84Transform.transform( point );
-    updateGpsDistanceStatusMessage();
+    updateGpsDistanceStatusMessage( true );
   }
   catch ( QgsCsException & )
   {
 
   }
 }
 
-void QgsGpsInformationWidget::updateGpsDistanceStatusMessage()
+void QgsGpsInformationWidget::updateGpsDistanceStatusMessage( bool forceDisplay )
 {
   if ( !mNmea )
     return;
 
+  if ( !forceDisplay )
+  {
+    // if we aren't forcing the display of the message (i.e. in direct response to a mouse cursor movement),
+    // then only show an updated message when the GPS position changes if the previous forced message occurred < 2 seconds ago.
+    // otherwise we end up showing infinite messages as the GPS position constantly changes...
+    if ( mLastForcedStatusUpdate.hasExpired( 2000 ) )
```

Comment:
Can you create a constant with this 2000 instead of this magic number?
BTW, do you know why it's exactly 2 seconds?
Thanks.

Reply:
It's just the default timeout for timed status bar messages -- but yes, I'll make it a constant